llvm/lib/Target/PowerPC/PPCAsmPrinter.cpp

```diff
@@ -1539,7 +1528,9 @@ void PPCAsmPrinter::emitInstruction(const MachineInstr *MI) {
     // with an immediate operand having the MO_TPREL_FLAG. Such an instruction
     // does not otherwise arise.
     const MachineOperand &MO = MI->getOperand(2);
-    if ((MO.getTargetFlags() & PPCII::MO_TPREL_FLAG) != 0) {
+    if (MO.getTargetFlags() == PPCII::MO_TPREL_FLAG ||
```

Comment:
Perhaps the same here, we could pull out `MO.getTargetFlags()` due to multiple accesses.

Reply:
Done